rup-projects-core/src/main/java/com/core/rupprojectscore/resource/UseCaseResource.java

```diff
@@ -0,0 +1,60 @@
+package com.core.rupprojectscore.resource;
+
+import com.core.rupprojectscore.dto.UseCaseDto;
+import com.core.rupprojectscore.service.UseCaseService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+
+@RestController
+@RequestMapping(value = UseCaseResource.USE_CASES, produces = MediaType.APPLICATION_JSON_VALUE)
+@RequiredArgsConstructor
+public class UseCaseResource {
+
+    public static final String USE_CASES = "/usecases";
+    public static final String PRIORITIZE_USE_CASES = "usecases/prioritize";
+    public static final String USE_CASE_ID = "{id}";
+
+    private final UseCaseService service;
+
+    @PostMapping
+    public ResponseEntity<UseCaseDto> createUseCase(final @RequestBody UseCaseDto dto) {
+        return ResponseEntity.ok(this.service.createUseCase(dto));
+    }
+
+    @GetMapping
+    public List<UseCaseDto> openUseCases() {
+        return this.service.openUseCases();
+    }
+
+    @PutMapping(USE_CASE_ID)
+    public ResponseEntity<UseCaseDto> updateUseCase(final @PathVariable("id") Long id, final @RequestBody UseCaseDto dto) {
+        dto.setId(id);
+        return ResponseEntity.ok(service.updateUseCase(dto));
+    }
+
+    @PostMapping(path = PRIORITIZE_USE_CASES)
+    public ResponseEntity prioritizeUseCases(final @RequestBody List<UseCaseDto> useCasesDto) {
+        service.prioritizeUseCases(useCasesDto);
+        return ResponseEntity.ok().build();
+    }
```

Comment:
para cumplir con el diagrama de colaboracion necesito este metodo en el Resourse, no se que tan correcto es hacer un POST usecases/prioritize 

con un body que es un listado de usecaseDto

Reply:
En cualquier caso sería un PUT, porque sería para actualizar las prioridades de los recursos

Reply:
REvisando el diagrama de colaboración, ¿Lo que actualizamos no es el useCase con el valor nuevo de prioridad?
![image](https://user-images.githubusercontent.com/4791544/93000002-9d4ebe80-f525-11ea-8c58-f3809c63371d.png)


Yo lo veo como un simple put. Lo que pasa es que habría que llamar al endpoint de otra forma, algo en plan:

put /usecases/x/priority

Reply:
Sii es que tambien cambie el diagrama de colaboracion tenemos esta especificacion 

![image](https://user-images.githubusercontent.com/16058725/93000212-18fd3b00-f527-11ea-8072-a38a166b1618.png)

El usuario va intoduciendo valores de priority para los casos de uso, y cuando acaba de hacerlo le da un boton y confirma la introduccion de esos valores, 

Si lo hacemos como en el viejo diagrama que pasaste tu la foto, estamos haciendo el cambio en DB cada vez que el usuario añade un valor de prioridad a un caso de uso, y si al final decide cancelar tendriamos que volver a borrar esos valores de la DB. 

Y por otro lado si al final del front tenemos una lista de useCases id a los que settear la prioridad, 
y hacemos individualmente 
put /usecases/1/priority
put /usecases/2/priority

Esa logica estaria en el UseCasesService de Angular, que si migramos a otro front habria que re-hacerlo en ese front

Por eso segun la especificacion yo lo veo mas de este modo 
El usuario va setteando las prioridades y cuando decide confirmar es cuando se llama al priorizeUsecases 
put /usecases/priorization

y los actualizas de golpe y toda la logica del PriorizeUseCaseController del analisis queda en la parte de la API 

![image](https://user-images.githubusercontent.com/16058725/93000291-bb1d2300-f527-11ea-9762-51f2c874fe8c.png)

Reply:
Vale, entiendo lo que dices. Si segun nuestro diseño de la interfaz, el usuario pulsa un botón priorizar y acepta la priorización de varios casos de uso a la vez, entonces totalmente de acuerdo, es una única llamada porque el usuario aun no ha terminado el caso de uso hasta que acepta la priorización. 

En cuanto a la orden, seguría siendo un PUT sin más. Sólo que habría que sobrecargar el endpoint para que ejecute el de update para 1 solo elemento, y el de prioritize cuando sea una lista de elementos. Eso se puede hacer en spring con "params".
Te quedaría algo así:

Para el prioritize
`@PutMapping(params = { !id })`
PAra el update:
`@PutMapping(params = { id })`

se discrimina por si aparece o no una propiedad id en el body de la request.

También te funcionará por supuesto con PUT /usecases/prioritize pero esta notación juraría que ya no cumple REST

Reply:
![image](https://user-images.githubusercontent.com/16058725/93015150-70e38280-f5b7-11ea-9966-26076209259e.png)

sin el { !id } lo estaria pillando en el path de /usecases/
entiendo que seria similar a poner { !id }

Reply:
Cierto, puedes hacer put a la url base y pasarle los que quieres actualizar por el body. Aunque creo que no sigue el standar rest porque put es orientado a un unico recurso: PUT /items/1

Según he estado googleando, si el PUT te da problemas, puedes usar PATCH, que sigue siendo una actualización, y sería algo así como "actualizame los items que te pase", por lo que podría ser incluso mas consistente.

Hecha un ojo aquí, como te digo, parece que se decantan por patch e incluso comentan que put para varios elementos podría no funcionar:

https://stackoverflow.com/questions/32098423/rest-updating-multiple-resources-with-one-request-is-it-standard-or-to-be-avo

Te resuelvo el hilo, así que cuando lo tengas puedes mergearlo